### ISSUE TITLE
Change Request: Mobile App Feature Affinity for Veteran Status Card

### ISSUE THREAD

**Issue by mcommons10** [department-of-veterans-affairs/va-mobile-app#9480]
### 1.) Requestor's Name

Megan Commons

### 2.) Requestor's Email

megan.commons@oddball.io; megan.commons@va.gov

### 3.) VA Team

Iterate, Innovate, and Run

### 4.) What is the name of the project or change?

Mobile App Feature Affinity for Veteran Status Card

### 5.) Please explain what problem you are trying to solve.

The Iterate, Innovate, and Run (IIR) team is driving the Veteran Status experience both on the web and mobile app. While a longer term product strategy is being worked on, there is a short term need to first have feature affinity between the web and mobile app experience. Problem statements:
- Users have different experiences and different information between the web and mobile app.
- Mobile app users don’t understand why they don’t have access or can’t access their Proof of Status (no error/ineligibility messaging).
- VA Stakeholders (Melissa Rebstock) will not have a strong case to ask for legislative changes to card requirements if Vet Status does not contain an ID number that is not a SSN (adding the DoD ID). 

### 6.) Please explain how this change would work.

Desired User Outcomes:
- Mobile app users can see their DoD ID number on their Proof of Status
- Mobile app users who are ineligible for Proof of Status know why they are ineligible
- Mobile app users who can’t access their Proof of Status know why they can’t access
- Increase the number of Veterans that can view their Proof of Status (potential API change to reduce errors).

### 7.) Features Impacted

Proof of Veteran Status

### 8.) Please include all relevant documentation links for this change request.

Product Outline: https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/veteran-status/v2-IIR/mobile-app-feature-affinity-product-outline.md
Figma Designs: https://www.figma.com/design/AobGKkBvIiDuYXR2QqkytC/IIR-Veteran-Status?node-id=641-83728&node-type=SECTION&t=zUC9O2dfoTPvfKCA-0
Logic for Error Messaging: https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/veteran-status/v2-IIR/ineligibility-error-messaging-logic.md
Github Product Folder (v1 was previous mobile app team work): https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/veteran-status

### 9.) Does your team have all the necessary materials (documents, designs, processes, analytics, APIs, etc) in place for Mobile to make these changes right now?

No, we are currently working on the mobile app designs and making a final decision on API usage.

### 10.) Has this change been discussed with VAHB POs (Rachel, Ryan, or Don)?

Not in great length, no. But we did kickoff a discussion in Slack: https://dsva.slack.com/archives/C018V2JCWRJ/p1724171474649329

### 11.) Has this change been agreed upon by all impacted VA Stakeholders?

From the IIR side of things, yes. Need mobile app team input.

### 12.) What is your ideal due date for this change request?

10/15/2024

**Comment by rtwell** [comment]
What legislative requirements does this feature need to meet? AFAIK the VIC is already not meeting some of them by not charging veterans for this service. 

@DonMcCaugheyUSDS will have follow up questions re: API

**Comment by mcommons10** [comment]
@rtwell - There is an ongoing effort being made to ask for an amendment on legislation requirements (Melissa Rebstock has been point on this). Our goal is to have as many of the requirements on the Vet Status Card as possible without requiring manual Veteran actions. Here is what is required by legislation to have on the VIC: 
- Name of the Veteran 
- Photograph of the Veteran 
- Explanation that the card is not proof of any benefits
- contains an ID number that is not a SSN (adding the DoD ID will address this)

Our main focus is to take this product that already exists and have the same experience between web and mobile app, not to drive the decommissioning of the VIC (that is Melissa Rebstock's wheelhouse).

**Comment by DonMcCaugheyUSDS** [comment]
Looking at the [Ineligibility Error Messaging Logic](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/veteran-status/v2-IIR/ineligibility-error-messaging-logic.md) doc, I see that the current business logic for veteran status and ineligibility errors lives in the web client code in the vets-website repo.

We need to move this business logic out of the web client into the vets-api backend so it is centralized in one place that can be shared by both the web client and the mobile client.  This will prevent problems in the future where web and mobile show different results or errors.

**Comment by rtwell** [comment]
@mcommons10 I'm trying to reconcile the problem of 
>The Veteran ID Card (VIC) cannot be decommissioned until both the web and mobile app Proof of Status meet legislative requirements (DoD ID number needs to be added).
Is this a problem for this team to solve?

**Comment by mcommons10** [comment]
@rtwell - While we are not in control of decommissioning of the VIC, we are in control of making updates to Vet Status that will provide Melissa a better persuasive case when asking for legislative changes. So I guess that problem statement could be reworded to say: Melissa will not have a strong case to ask for legislative changes if Vet Status doesn’t contains an ID number that is not a SSN (adding the DoD ID).

**Comment by mcommons10** [comment]
@DonMcCaugheyUSDS - thanks for the direction on this. We'll plan on having a SPIKE ticket in a coming sprint to identify what this would actually take to implement and make available. I'll keep you posted.

**Comment by mcommons10** [comment]
@rtwell - done! thanks for the input

**Comment by mcommons10** [comment]
Thanks @rtwell! Sounds like a plan. I'll reach out to @ajsarkar28 in Slack this week and come up with a plan.

**Comment by rtwell** [comment]
great tyvm! just a reminder to keep convos in public channels so folks can follow along—thank you!! exciting to see this feature evolve.

**Comment by rtwell** [comment]
just wanted to drop a couple clarifying notes as we all figure out how mobile app + experience teams work together:
1. @Hallm13 I jumped the gun pinging Ameet directly here. In future work, we'll make sure to include you, and then you can delegate as needed. 
2. @JeffBarnesUSDS and @mcommons10 this is your initiative, the mobile team will take a back seat here and you all should lead / drive this work. we'll leave it up to you all to initiate meetings/reviews/communications as needed. We know it's a little undefined and messy right now, we'll figure out an effective cadence as we go.

**Comment by mcommons10** [comment]
Thanks @rtwell, that sounds great to me. I plan on starting a thread in the va-mobile-app channel this week to get things moving. Appreciate the clarity.

**Comment by kellylein** [comment]
[Here](https://github.com/department-of-veterans-affairs/va.gov-team/issues/94373) is the Design Intent ticket for this work.

**Comment by DonMcCaugheyUSDS** [comment]
Linking the mobile global team epic [Update Veteran Status Date of Birth Data to Limit Personal Information #9865](https://github.com/department-of-veterans-affairs/va-mobile-app/issues/9865).

We can start developing the mobile app work for the improved error messages part of this change when we have shared backend code to work from.

**Comment by mcommons10** [comment]
Thanks @DonMcCaugheyUSDS! As an update, we have moved the backend code from vets-website to vets-api but we are currently working on changing APIs (moving to the [Veteran Service History & Eligibility API](https://developer.va.gov/explore/api/veteran-service-history-and-eligibility/docs?version=current)) and once that is complete, error messaging can be worked on.

**Comment by BeccaTupaj** [comment]
@mcommons10 @nataliedavied @parkerbees Do you know the epic that is tracking the work on the Veteran Status Card error messaging? 

I'd like to close this change request ticket if the work is being tracked elsewhere.

**Comment by NatalieDavied** [comment]
@BeccaTupaj The MFS team is using this [epic](https://github.com/department-of-veterans-affairs/va-mobile-feature-support/issues/83) to track the work on the Veteran Status Card error messaging.

**Comment by BeccaTupaj** [comment]
Thanks @NatalieDavied. I will close this Change Request and future work can be tracked in the ticket Natalie linked above.

**Comment by BeccaTupaj** [comment]
@mcommons10 Please let me know if there is additional work involved in this change request that is not yet being worked on by the MFS team. 
cc: @rtwell
